File: testdata/go/interfaces/analyse.go
package sample

import (
	"context"
	"io"
)

type Reader interface {
	Read(p []byte) (int, error)
}

type PaymentProvider interface {
	io.Closer
	Reader
	Charge(ctx context.Context, req ChargeRequest) (ChargeResponse, error)
	Close() error
}

File: testdata/dart/interfaces/analyse.dart
abstract class Reader {
  String read(String id);
}

abstract class PaymentProvider extends Reader implements Logger, Chargeable {
  Future<void> charge(String id);
  String format(int x);
  void withBody() {}
}

class Worker extends Reader {
  @override
  String read(String id) => id;
}
